File: src/scripts/audit_epl.py
import os
import pandas as pd
import glob

data_dir = "data/EPL"
csv_files = glob.glob(os.path.join(data_dir, "*.csv"))

print("--- Phase A: Dataset Audit (EPL Historical Data) ---")
print(f"Found {len(csv_files)} seasons in {data_dir}")

total_matches = 0
missing_data_summary = pd.DataFrame()

all_dfs = []

for file in sorted(csv_files):
    df = pd.read_csv(file)
    season_name = os.path.basename(file).replace('.csv', '')
    matches_in_season = len(df)
    total_matches += matches_in_season
    print(f"\nSeason: {season_name} - Matches: {matches_in_season}")
    
    # Check for critical columns
    columns = set(df.columns)
    print(f"Columns found: {len(columns)}")
    
    # Check for odds coverage
    has_b365 = 'B365H' in columns
    has_asian = 'B365AHH' in columns or 'AHh' in columns
    has_ou = 'B365>2.5' in columns or 'BbMx>2.5' in columns
    
    print(f"Moneyline Coverage: {'YES' if has_b365 else 'NO'}")
    print(f"Asian Handicap Coverage: {'YES' if has_asian else 'NO'}")
    print(f"Over/Under Coverage: {'YES' if has_ou else 'NO'}")

    all_dfs.append(df)

if all_dfs:
    combined = pd.concat(all_dfs, ignore_index=True)
    print(f"\nTotal Matches across all seasons: {len(combined)}")
    
    missing = combined.isnull().sum()
    print("\nTop Missing Columns:")
    print(missing[missing > 0].sort_values(ascending=False).head(10))

File: src/scripts/audit_epl.ts
import * as fs from 'fs';
import * as path from 'path';

const dataDir = path.join(process.cwd(), 'data/EPL');

function parseCSV(content: string) {
  const lines = content.split('\n').filter(l => l.trim().length > 0);
  if (lines.length === 0) return { headers: [], rows: [] };
  const headers = lines[0].split(',').map(h => h.trim());
  const rows = lines.slice(1).map(l => l.split(',').map(c => c.trim()));
  return { headers, rows };
}

async function main() {
  console.log('--- Phase A: Dataset Audit (EPL Historical Data) ---');
  
  if (!fs.existsSync(dataDir)) {
    console.error(`Directory not found: ${dataDir}`);
    return;
  }

  const files = fs.readdirSync(dataDir).filter(f => f.endsWith('.csv')).sort();
  console.log(`Found ${files.length} seasons in ${dataDir}\n`);

  let totalMatches = 0;
  const allHeaders = new Set<string>();

  for (const file of files) {
    const content = fs.readFileSync(path.join(dataDir, file), 'utf-8');
    const { headers, rows } = parseCSV(content);
    
    headers.forEach(h => allHeaders.add(h));
    totalMatches += rows.length;

    const hasB365 = headers.includes('B365H');
    const hasAsian = headers.includes('B365AHH') || headers.includes('AHh');
    const hasOU = headers.includes('B365>2.5') || headers.includes('BbMx>2.5');

    console.log(`Season: ${file.replace('.csv', '')} - Matches: ${rows.length}`);
    console.log(`  Columns found: ${headers.length}`);
    console.log(`  Moneyline Coverage: ${hasB365 ? 'YES' : 'NO'}`);
    console.log(`  Asian Handicap Coverage: ${hasAsian ? 'YES' : 'NO'}`);
    console.log(`  Over/Under Coverage: ${hasOU ? 'YES' : 'NO'}\n`);
  }

  console.log(`Total Matches across all seasons: ${totalMatches}`);
}

main().catch(console.error);
